hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmSnapshotManager.java

```diff
@@ -905,7 +905,37 @@ public void snapshotLimitCheck() throws IOException, OMException {
   }
 
   public void decrementInFlightSnapshotCount() {
-    inFlightSnapshotCount.decrementAndGet();
+    /*
+     * There is a race condition here because of which
+     * inFlightSnapshotCount could become negative sometimes.
+     *
+     * When we get a call back from Ratis on notifyLeaderReady,
+     * we assume that all the pending transactions (from RaftLog)
+     * are applied to the SateMachine and we go a head and reset the
+     * inFlightSnapshotCount to 0. The expectation here is that after
```

Comment:
We should fix the accounting logic of `inFlightSnapshotCount`.
- When a request is submitted, increment the count;
- When the same request is completed/falied/cancelled, decrement the count.

It should not rely on an external condition to reset the count.